### ISSUE TITLE
Enso builtins don't expose Enso methods

### ISSUE THREAD

**Issue by JaroslavTulach** [enso-org/enso#11589]
Quite surprisingly Enso builtins don't expose Enso methods to `org.graalvm.polyglot.Value`. That results in quite surprising situations:

- cannot call `Ref.get`: https://github.com/enso-org/enso/pull/11577/files#r1848224116
- cannot call `File.path` on [EnsoFile](https://github.com/enso-org/enso/pull/11530/files#r1838708041)
- cannot call `to_text` on `Vector` - discovered while working on #11569

Looks like _Enso builtin types_ do **not expose** their Enso methods to `org.graalvm.polyglot.Value`. Let's write `ValuesGenerator` based test and verify all Enso builtins expose their (public) Enso methods properly.

**Comment by Akirathan** [comment]
The following test obviously fails:
https://github.com/enso-org/enso/blob/f9a00c6a2a7612af536550c75c7df94ce52410af/engine/runtime-integration-tests/src/test/java/org/enso/interpreter/test/builtins/BuiltinsExposeMethodsTest.java#L41-L51

`ref.canInvokeMember("get")` sends `isMemberInvocable("get")` message to the interop library implementation of [Ref](https://github.com/enso-org/enso/blob/04075c5ed74075f5113a6e7a06a8d1fec0d25918/engine/runtime/src/main/java/org/enso/interpreter/runtime/data/Ref.java). `Ref` does not expose such message and, therefore, fails.

Interop protocol for Atom changed recently in https://github.com/enso-org/enso/pull/11217 and its `getMembers` message implementation says in the docs:
https://github.com/enso-org/enso/blob/04075c5ed74075f5113a6e7a06a8d1fec0d25918/engine/runtime/src/main/java/org/enso/interpreter/runtime/data/atom/Atom.java#L156-L160

The obvious solution for this issue is to make the interop protocol for every builtin consistent with the protocol of [Atom](https://github.com/enso-org/enso/blob/04075c5ed74075f5113a6e7a06a8d1fec0d25918/engine/runtime/src/main/java/org/enso/interpreter/runtime/data/atom/Atom.java). More specifically, make these interop messages consistent:
- `getMembers`
- `hasMembers`
- `isMemberReadable`
- `isMemberInvocable`
- `isMemberInternal`
- `getMetaObject`
- `hasMetaObject`

**Comment by JaroslavTulach** [comment]
> * Should every `EnsoContext` has a separate instance of `Builtins`, or can they be shared?

No, they cannot be shared. Each `EnsoContext` should currently have its own instance of `Builtins`.

> * How does this relate to the static field `Builtins.loadedBuiltinConstructors`?

Right, how? I'd like to know the answer too. Preferably without me using the debugger.

**Comment by Akirathan** [comment]
> > * Should every `EnsoContext` has a separate instance of `Builtins`, or can they be shared?
> 
> No, they cannot be shared. Each `EnsoContext` should currently have its own instance of `Builtins`.
> 
> > * How does this relate to the static field `Builtins.loadedBuiltinConstructors`?
> 
> Right, how? I'd like to know the answer too. Preferably without me using the debugger.

@JaroslavTulach Answer provided in https://github.com/enso-org/enso/pull/11861#issuecomment-2551236755
